ui/radar/radar.js

```diff
@@ -0,0 +1,167 @@
+      }
+    }
+    // removing new combatant
+    r = e.detail.logs[i].match(/04:........:Removing combatant (.*)\. (?:.*) Max HP: (\d+)\./);
+    if (r) {
+      for (let i = 0; i < targetMonsters.length; i++) {
+        let monster = targetMonsters[i];
+        let arrow_id = 'arrow-' + monster['id'];
+        if (monster['dom'] && r[1].match(monster['name'])) {
+          if (monster['cur_hp'] > 50)
+            continue;
+          monster['dom'].remove();
+          targetMonsters.splice(i, 1);
+          break;
+        }
+      }
+    }
+    // debug:
+    r = e.detail.logs[i].match(/:record pos/);
+    if (r) {
+      let Vt = new Point2D(playerPos['x'], playerPos['y']);
+      target.push(Vt);
+      console.log(playerPos);
+      console.log(target);
+    }
+    r = e.detail.logs[i].match(/:reset pos/);
+    if (r) {
+      target = [];
+      console.log(playerPos);
+      console.log(target);
+    }
+  }
+});
+
+document.addEventListener('onPlayerChangedEvent', function(e) {
+  playerPos['x'] = e.detail.pos.x;
+  playerPos['y'] = -e.detail.pos.y;
+  let tr;
+  for (let i = 0; i < targetMonsters.length; i++) {
+    let monster = targetMonsters[i];
+    let arrow_id = 'arrow-' + monster['id'];
+    if (monster['dom'] == null) {
+      // add dom
+      let tr = document.createElement('tr');
+      let th = document.createElement('th');
+      let img = document.createElement('img');
+      img.setAttribute('id', arrow_id);
+      img.setAttribute('src', 'arrow.png');
+      img.setAttribute('class', 'radar-image-40');
+      th.appendChild(img);
+      th.setAttribute('style', 'max-width: 100px');
+      tr.appendChild(th);
+      th = document.createElement('th');
+      let text = document.createElement('div');
+      text.innerHTML = monster['name'];
+      th.appendChild(text);
+      tr.appendChild(th);
+      $('#radar-table').prepend(tr);
+      monster['dom'] = tr;
+    } else {
+      tr = monster['dom'];
+    }
+    // caculate rotate
+    let Vp = new Point2D(e.detail.pos.x, -e.detail.pos.y); // player vector
+    let Vt = monster['pos'];
+    let Vdelta = new Point2D(Vt.x - Vp.x, Vt.y - Vp.y);
+    if (tr)
+      tr.childNodes[1].innerHTML = monster['name'] + ' ' + length(Vdelta).toFixed(2) + 'm';
+    let Tdelta = Math.acos(theta(Vdelta, new Point2D(1, 0)));
+    if (Vdelta.y < 0)
+      Tdelta = -Tdelta;
+    Tdelta += Math.PI - e.detail.rotation;
+    let t = -Tdelta * 180 / Math.PI;
+    $('#' + arrow_id).rotate(t);
+  }
+  // debug:
+  // let Vp = new Point2D(e.detail.pos.x, -e.detail.pos.y);
+  // let Vt = target[0];
+  // if(Vt != null){
+  //   let Vdelta = new Point2D(Vt.x - Vp.x, Vt.y - Vp.y);
+  //   let Tdelta = Math.acos(theta(Vdelta, new Point2D(1, 0)));
+  //   if(Vdelta.y < 0)
+  //     Tdelta = -Tdelta
+  //   Tdelta += Math.PI - e.detail.rotation;
+  //   let t = -Tdelta*180/Math.PI;
+  //   $('#test-demo').rotate(t);
+  // }
+});
+
+document.addEventListener('onZoneChangedEvent', function(e) {
```

Comment:
Can you add a `UserConfig.getUserConfigLocation('radar', etc....` bit here, like there is elsewhere in the code? That will let users override this with `cactbot/user/radar.js` or `.css` files.

Reply:
Actually I'm not sure how the `UserConfig` works, I add a `UserConfig.getUserConfigLocation` to the end of `radar.js` but not sure whether this will overwrite files.

Reply:
Calling getUserConfigLocation(radar) automatically includes `radar.css` and `radar.js` from the cactbot/user folder.  If there are things you don't want to set up or start until the full options have been set (like maybe the user wants to override something), most cactbot overlays do all of their setup (like adding event listeners, etc) inside the callback that it gets.  Pullcounter is a pretty straightforward example if you want to look at that.

You can try this out yourself by creating a `cactbot/user/radar.js` file and adding a `console.log('custom radar user file');` line in it.

I'm not sure what you mean about overwriting files.

Reply:
Okay I add the user config file as `user/radar-example.js`